umbraco/Umbraco-CMS
# Use ImageSharps Image.Identify for dimension extraction

### Prerequisites


### Description
This replaces the `Image.Load(stream)` call in the `IImageDimensionExtractor` implementation with `Image.Identify(stream)`, which is more memory and CPU efficient: https://docs.sixlabors.com/articles/imagesharp/imageformats.html#metadata-only-decoding.

I've also cleaned up related bits, notably:
- Removing `ExifImageDimensionExtractor`, as the same is already done by ImageSharp
- Removing `ImageSize` struct, as we can use the built-in `Size` struct
- Move the default size fallback outside of the `IImageDimensionExtractor` implementation

### Testing
The changes impact the automatic population of the `umbracoWidth` and `umbracoHeight` properties (part of the built-in Image media type):
- Upload an image with a valid file extension (e.g. `.jpg`) to the media section: the properties should reflect the correct image size;
- Remove the file from the media item and upload a dummy text file with some contents and the same file extension as the image (e.g. `.jpg`): the default sizes of 200x200 should be set, as the file extension is a valid image extension, but it can't parse the dimensions from the 'image';
- Remove the file again and upload the dummy text file with a `.txt` extension: the dimensions should be cleared, as it's not a valid image extension.
